PolyEngine/Engine/TimeSystem.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <chrono>
+#include <Core.hpp>
+
+
+namespace Poly
+{
+	class ENGINE_DLLEXPORT TimeSystem : public BaseObject<>
+	{
+	public:
+
+		TimeSystem();
+		~TimeSystem() {};
+
+		float Update();
+		float GetDeltaTime() const { return DeltaTime.count(); };
+		float GetPausableTimer() const { return PausableTimer.count(); };
+		float GetTimer() const { return Timer.count(); }
+
```

Comment:
Don't know how much do we care, but game time should actually be of `double` type. Float's precision is simply too low. After the game has been running for a day the precision drops to 7.8 milliseconds, and after a week to 62.5 milliseconds! (admittedly that's a pretty rare case) And then you've got stuff like occasional catastrophic cancellations when calculating delta time.